plenum/server/request_handlers/txn_author_agreement_handler.py

```diff
@@ -54,24 +60,44 @@ def _decode_state_value(self, encoded):
             return value, last_seq_no, last_update_time
         return None, None, None
 
-    def _validate_add_taa(self, request):
+    def _validate_add_taa(self, request: Request, req_pp_time: int):
         if request.operation.get(TXN_AUTHOR_AGREEMENT_TEXT) is None:
             raise InvalidClientRequest(request.identifier, request.reqId,
-                                       "Cannot create a transaction author agreement without a '{}' field."
+                                       "Cannot create transaction author agreement without a '{}' field."
                                        .format(TXN_AUTHOR_AGREEMENT_TEXT))
 
         if request.operation.get(TXN_AUTHOR_AGREEMENT_RETIREMENT_TS) is not None:
             raise InvalidClientRequest(request.identifier, request.reqId,
-                                       "Cannot create a transaction author agreement with a '{}' field."
+                                       "Cannot create transaction author agreement with a '{}' field."
                                        .format(TXN_AUTHOR_AGREEMENT_RETIREMENT_TS))
 
+        ratification_ts = request.operation.get(TXN_AUTHOR_AGREEMENT_RATIFICATION_TS)
+        if ratification_ts is None:
+            raise InvalidClientRequest(request.identifier, request.reqId,
+                                       "Cannot create transaction author agreement without a '{}' field."
+                                       .format(TXN_AUTHOR_AGREEMENT_RATIFICATION_TS))
+
+        if ratification_ts > req_pp_time:
```

Comment:
Do we consider some Delta (for a possible clock skew) here?

Reply:
Good question. We might need it, but it wasn't in requirements. Also, I think this shouldn't be a problem since production use case most probably will look like:
- create TAA txn with ratification time current or earlier
- pass TAA txn to trustees so they sign it (this process most probably will take at least several minutes)
- submit signed TAA txn - with ratification date already well in the past

Also, when author of TAA txn is trustee and he wants to send txn immediately he can put ratification date into the past himself (which is also done by integration tests).